### DISCUSSION TITLE
Store Integration Queries: Paging, Sorting, and Complex Search Handling

### DISCUSSION THREAD

**Discussion by leinardi** [dropbox/Store#567] [Q&A]
Hi, in the app I'm developing, I need to display a list of Posts and a list of Comments. I can retrieve these posts and comments from an API, and I can paginate the results and choose the sorting order. I was wondering if it is possible to achieve the following things using Store:

Integrate with the Android Jetpack Paging 3 library to properly handle lazy lists.
Allow local modifications to some of the items, such as upvoting or downvoting a post/comment, or adding a comment to a post. These actions will also be sent to the backend via API, but I would like to update the app's UI first with the local changes to provide a better user experience.
Handle complex search queries: The API allows me to fetch a list of Posts and Comments, sorting and filtering them using various arguments. How would this work with Store if, for example, I change the sorting parameter of the Post list from "Newer" to "Older"?

**Comment by matt-ramotar** [comment]
Hey @leinardi - Sorry to be slow. Thanks for reaching out. We are actively thinking about how to support paging and complex operations. I'll have more time in Q4 to tackle this. I have a RFC in draft and am aiming to share it more broadly in the next few weeks.

Related:
- #250 
- #550

**Comment by matt-ramotar** [comment]
Hey @leinardi - I merged #550 today. The PR has a design doc. Let me know what you think
